server/rooms.ts

```diff
@@ -1756,11 +1783,11 @@ export const Rooms = {
 
 		if (privacySetter.size) {
 			if (battle.forcePublic) {
-				room.settings.isPrivate = false;
+				void room.makePublic();
```

Comment:
why are we not propagating the promise here and below?

Reply:
Because propagating clutters up a lot of unrelated code with `async`s (including a `setTimeout` in ladder that would look a lot less readable). It's definitely possible to propagate it but I don't forsee it causing problems.

Reply:
I strongly disagree with ever `void`-ing promises (at the very least they need to be propagated to the top level and cases of ignoring it at the top level should be accompanied with a comment justifying it) and I really which PS didn't normalize this and teach developers it is OK to sacrifice correctness at the alter of 'avoiding clutter', but I've lost that battle.

Reply:
Unnecessarily propagating promises also makes the code less parallel and ultimately slower. It looks like the async-ness here comes from `rename`, which I'm pretty sure doesn't need to be propagated outside of the relevant roomlog function at all (the relevant roomlog functions needs to be able to handle the rename without race conditions whether or not the Promise is propagated). But I suppose you're not interested in having this argument.

Reply:
If you believe changing the argument in favor from 'avoiding clutter' to 'performance reasons' is going to convince me that making something untestable and incorrect is going to satisfy me then yes, I'm not interested in having that argument.

Feel free to ignore this, of course. I am never going to approve/merge code that doesn't propagate promises, but you do you.

Reply:
(I'm happy to propagate it or not once you two figure this out - I was just going off PS' existing code style choices)

Reply:
It doesn't seem like I'm going to understand pre's perspective; he's explained it to me a few times and it's never clicked for me. Let's just put the `void` inside `this.rename` (change the last `return` to `void`).